### PR TITLE
refactor: yarn workspaces, cross-fetch, tsup

### PR DESCRIPTION
**Pull request by jiito**
# Changes 
- refactored the repo to use yarn workspaces 
     eventually we might want to use something like turborepo or lerna to orchestrate the packages. 
     allows for testing of local dependencies easily in different environments 
- created example next, vite, and express packages for testing and demonstration purposes 
- using tsup to build the core package (standard from other ts packages I have looked at)

### CODE REVIEW

File: tsconfig.json
Comment (jiito) [inline tsconfig.json:1]: This was moved the the `/package/core` directory

File: packages/core/src/index.ts
Comment (jiito) [inline packages/core/src/index.ts:1]: @jeffzwang I moved all the named exports to a single export of types at the bottom for clarity.

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -95,29 +96,45 @@ export default class Metaphor {
...
+  SearchOptions,
+  Result,
+  SearchResponse,
+  FindSimilarOptions,
+  DocumentContent,
+  GetContentsResponse,
+};
+
+export default Metaphor;
+
```
Comment (jiito) [inline packages/core/src/index.ts:137]:
```suggestion
// Necessary to allow for ergonomic 'const metaphor = require('metaphor-node') in CommonJS
```

**Review by jeffzwang — APPROVED** [review]
thanks @jiito !

**Pull request merged**